src/lib/sort/sort.ts

```diff
@@ -54,20 +56,35 @@ export interface Sort {
   direction: SortDirection;
 }
 
-// Boilerplate for applying mixins to MatSort.
-/** @docs-private */
-export class MatSortBase {}
-export const _MatSortMixinBase: HasInitializedCtor & CanDisableCtor & typeof MatSortBase =
-    mixinInitialized(mixinDisabled(MatSortBase));
+export interface MatSortContainer<T> {
+  /** Stream that emits an event when any sortable's state may have changed. */
+  _stateChanges: Observable<void>;
+
+  /** Registers the sortable so that it can be sorted. */
+  register(sortable: MatSortable): void;
```

Comment:
Looking over the code, I'm starting to now think we _should_ move some of this to the cdk.

Reply:
+1 Let's chat